tests/scripts/task_config_build_gpu.sh

```diff
@@ -23,12 +23,28 @@ mkdir -p "$BUILD_DIR"
 cd "$BUILD_DIR"
 cp ../cmake/config.cmake .
 
+# GoogleTest uses a Live-at-Head philosophy:
+# https://github.com/google/googletest#live-at-head
+# therefore we need to grab a specific hash and update it
+# periodically to match the head of the repo
+repo_url="https://github.com/google/googletest"
+repo_revision="830fb567285c63ab5b5873e2e8b02f2249864916"
+archive_name="${repo_revision}.tar.gz"
+archive_url="${repo_url}/archive/${archive_name}"
+archive_folder="googletest-${repo_revision}"
+archive_hash="10f10ed771efc64a1d8234a7e4801838a468f8990e5d6d8fcf63e89f8d1455c4f9c5adc0bb829669f381609a9abf84e4c91a7fdd7404630f375f38fb485ef0eb"
+
+curl -sL "${archive_url}" -o "${archive_name}"
+echo "$archive_hash" ${archive_name} | sha512sum -c
+tar xf "${archive_name}"
```

Comment:
It is necessary to have google tests source code to build them for the device. In the CI we could use system GTest, but I didn't want to change the build mechanism in the CMake files, because I think it is better not to use system gtest but build it for the target device. 
This is why we need to download gtest source code in the build directory and build `tvm_runtime` with it.

Reply:
The downloading is fine but this should be moved into the GPU docker image (https://github.com/apache/tvm/blob/main/docker/Dockerfile.ci_gpu) so we don't have to download it fresh each time. You'll have to send a separate PR for the docker changes, then we can update the images to use the new one, then once that's done we can merge this PR